### PR TITLE
TECH-2231 Adds a spec helper to stenotype

### PR DESCRIPTION
**Pull request by KapitonovRoman**

### CODE REVIEW

File: lib/stenotype/test/matchers.rb [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        def multiple_events?
+          matching_events.size > 1
+        end
+
+        def matching_event
+          matching_events.first
+        end
+
+        def stringify_keys(hash)
+          Hash[hash.map { |key, value| [key.to_s, value] }]
```
Comment (glupo) [inline lib/stenotype/test/matchers.rb:59]: `hash.traansform_keys(&:to_s)` ?

File: lib/stenotype/test/matchers.rb [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+          buffer_before_emit = stenotype_event_buffer.dup
+          @emitting_event_block.call
+          buffer_after_emit = stenotype_event_buffer.dup
+
+          diff = buffer_after_emit - buffer_before_emit
+          @matching_events = diff.select { |event| event["name"] == expected_event_name.to_s }
+
+          partial_matchers << EventEmitted.new(@matching_events, expected_event_name)
+          partial_matchers << EventHasExpectedArguments.new(matching_events, @arguments_should_include) if should_validate_events_count?
+          partial_matchers << DiffSizeMatchesExpectation.new(matching_events, @matching_events_count) if should_validate_attributes?
```
Comment (glupo) [inline lib/stenotype/test/matchers.rb:95]: Shouldn't `matching_events` be instance variable here?

File: lib/stenotype/test/matchers.rb [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+          @emitting_event_block = emitting_event_block
+
+          buffer_before_emit = stenotype_event_buffer.dup
+          @emitting_event_block.call
+          buffer_after_emit = stenotype_event_buffer.dup
+
+          diff = buffer_after_emit - buffer_before_emit
+          @matching_events = diff.select { |event| event["name"] == expected_event_name.to_s }
+
+          partial_matchers << EventEmitted.new(@matching_events, expected_event_name)
```
Comment (glupo) [inline lib/stenotype/test/matchers.rb:93]: Why are we using instance variable here, but memoized method for two below?

File: lib/stenotype/test/matchers.rb [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        def failure_message
+          "expected to see #{expected_count} event(s) but got #{matching_events.count} event(s)."
+        end
+
+        def matches?
+          matching_events.count == expected_count
+        end
+      end
+
+      class EventHasExpectedArguments
```
Comment (glupo) [inline lib/stenotype/test/matchers.rb:26]: Feels like this matcher could be  `diffable?`

File: lib/stenotype/test/matchers.rb (hunk 3 above)
Comment (KapitonovRoman) [reply lib/stenotype/test/matchers.rb:93]: ooops :)

File: lib/stenotype/test/matchers.rb (hunk 4 above)
Comment (KapitonovRoman) [reply lib/stenotype/test/matchers.rb:26]: won't fix

**Pull request merged**